crates/accelerate/src/consolidate_blocks.rs

```diff
@@ -28,12 +25,21 @@ use qiskit_circuit::imports::{QI_OPERATOR, QUANTUM_CIRCUIT};
 use qiskit_circuit::operations::{ArrayType, Operation, Param, UnitaryGate};
 use qiskit_circuit::packed_instruction::PackedOperation;
 use qiskit_circuit::Qubit;
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+use smallvec::smallvec;
 
 use crate::convert_2q_block_matrix::{blocks_to_matrix, get_matrix_from_inst};
 use crate::euler_one_qubit_decomposer::matmul_1q;
 use crate::nlayout::PhysicalQubit;
 use crate::target_transpiler::Target;
-use crate::two_qubit_decompose::TwoQubitBasisDecomposer;
+use crate::two_qubit_decompose::{TwoQubitBasisDecomposer, TwoQubitControlledUDecomposer};
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Debug, FromPyObject)]
```

Comment:
I don't see any public uses of `DecomposerType`, so the enum can probably be kept private, right? Similarly, I didn't find where is the `FromPyObject` trait used, but I might have missed it.

Reply:
`FromPyObject` is used to call either the `TwoQubitBasisDecomposer` or the `TwoQubitControlledUDecomposer` from python, see this comment:
https://github.com/Qiskit/qiskit/pull/13884#discussion_r1967929870

Reply:
removing `pub` from the `enum` produces this warning:
`warning: type consolidate_blocks::DecomposerType is more private than the item consolidate_blocks`